base_custom_info/models/custom_info.py

```diff
@@ -124,3 +58,64 @@ def unlink(self):
         if res:
             info_values.unlink()
         return res
+
+    @api.one
+    @api.depends("custom_info_template_id",
+                 "custom_info_ids.value_id.template_id")
+    def _compute_dirty_templates(self):
+        """Know if you need to reload the templates."""
+        expected_properties = self.all_custom_info_templates().mapped(
+            "property_ids")
+        actual_properties = self.mapped("custom_info_ids.property_id")
+        self.dirty_templates = expected_properties != actual_properties
+
+    @api.multi
+    @api.returns("custom.info.value")
+    def get_custom_info_value(self, properties):
+        """Get ``custom.info.value`` records for the given property."""
+        return self.env["custom.info.value"].search([
+            ("model", "=", self._name),
+            ("res_id", "in", self.ids),
+            ("property_id", "in", properties.ids),
+        ])
+
+    @api.multi
+    def all_custom_info_templates(self):
+        """Get all custom info templates involved in these owners."""
+        return (self.mapped("custom_info_template_id") |
```

Comment:
This works as with normal python sets, joining all results and removing duplicates.